### ISSUE TITLE
Custom headers don't work as defined in docs

### ISSUE THREAD

**Issue by kieckhafer** [abdennour/react-csv#53]
I'm trying to us an array of objects as my data, and an array of objects to define custom headers. I'm doing this based on the example provided in the README, which doesn't seem to work (see bottom of this comment).

___

If I **_do not_** define custom headers, the data displays correctly with the original headers.

```
data = [
  {columnA: '1 - Test Data A', columnB: '1 - Test Data B' , columnC: '1 - Test Data C'},
  {columnA: '2 - Test Data A', columnB: '2 - Test Data B' , columnC: '2 - Test Data C'},
  {columnA: '3 - Test Data A', columnB: '3 - Test Data B' , columnC: '3 - Test Data C'}
];
```

Result:
```
"columnA","columnB","columnC"
"1 - Test Data A","1 - Test Data B","1 - Test Data C"
"2 - Test Data A","2 - Test Data B","2 - Test Data C"
"3 - Test Data A","3 - Test Data B","3 - Test Data C"
```
___

If I try to use custom headers via an array of objects, my data returned blank, and the headers are all returned as `[object Object]`. (This is the example in the README).

```
headers = [
  {label: 'columnA', key: 'Column A'},
  {label: 'columnB', key: 'Column B'},
  {label: 'columnC', key: 'Column C'}
];

data = [
  {columnA: '1 - Test Data A', columnB: '1 - Test Data B' , columnC: '1 - Test Data C'},
  {columnA: '2 - Test Data A', columnB: '2 - Test Data B' , columnC: '2 - Test Data C'},
  {columnA: '3 - Test Data A', columnB: '3 - Test Data B' , columnC: '3 - Test Data C'}
];
```

Result:
```
"[object Object]","[object Object]","[object Object]"
"","","","","","","",""
"","","","","","","",""
```
___

If I try to use custom headers via an array of strings, my headers display correctly, but no data is displayed.

```
headers = ['Column A', 'Column B', 'Column C'];

data = [
  {columnA: '1 - Test Data A', columnB: '1 - Test Data B' , columnC: '1 - Test Data C'},
  {columnA: '2 - Test Data A', columnB: '2 - Test Data B' , columnC: '2 - Test Data C'},
  {columnA: '3 - Test Data A', columnB: '3 - Test Data B' , columnC: '3 - Test Data C'}
];
```

Result:
```
"Column A","Column B","Column C"
"","","","","","","",""
"","","","","","","",""
```

___

From the `README`:

```
headers = [
  {label: 'First Name', key: 'firstname'},
  {label: 'Last Name', key: 'lastname'},
  {label: 'Email', key: 'email'},
];

data = [
  {firstname: 'Ahmed', lastname: 'Tomi' , email: 'ah@smthing.co.com'},
  {firstname:'Raed', lastname:'Labes' , email:'rl@smthing.co.com'} ,
  {firstname:'Yezzi', lastname:'Min l3b', email:'ymin@cocococo.com'}
];

<CSVLink data={data} headers={headers}>
    Download me
</CSVLink>
```

And this is the result: 
```
"[object Object]","[object Object]","[object Object]"
"","",""
"","",""
"","",""
```

**Comment by kieckhafer** [comment]
So it looks like bad timing on my part here.

I installed `react-csv` Monday morning, but didn't get any coding started until Monday evening.

When I installed it, `1.0.10` had not yet been released, and issue #31 was inside `1.0.10`. So I installed an older version, but then `1.0.10` was release before I got into coding, so I was reading the docs for the newer version once I actually got started working on it.

I just updated to `1.0.10`, and everything is working great.
